### PR TITLE
More workshed features

### PR DESCRIPTION
**Pull request by Pantocyclus**
Added substring matching and initials matching to config. 

Also supports switching out of DNA lab, as well as trainset -> CMC (and should in general switch to whatever workshed the user intended to after we run out of turns)

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -1,30 +1,57 @@
...
+      )
+  );
+
+  if (validWorksheds.length > 1) {
+    print(`Invalid Workshed: ${s} matches multiple worksheds! Matched:`);
+    validWorksheds.forEach(({ item }) => print(`${item}`));
+    throw new Error();
+  } else if (validWorksheds.length === 0) {
+    print(`Invalid Workshed: ${s} does not match any worksheds!`);
+    throw new Error();
```
Comment (Pantocyclus) [inline src/config.ts:51]: Pre-empting that grimoire actually catches the errors and throws its own errors, so `throw new Error(text)` would result in the text not getting printed.

Comment (horrible-little-slime) [reply src/config.ts]: Just make this a code comment

File: src/workshed/index.ts [hunk 2]
```diff
@@ -17,14 +18,20 @@ class GarboWorkshed {
...
     if (options.done) this.done = options.done;
     if (options.action) this.action = options.action;
+    this.minTurns = options.minTurns ?? 0;
   }
 
   canRemove(): boolean {
-    return this.done?.() ?? true;
+    return (
+      (this.done?.() ?? true) ||
+      estimatedTurns() - globalOptions.saveTurns <=
```
Comment (horrible-little-slime) [inline src/workshed/index.ts:32]: estimatedTurns() _should_ already be accounting for saveTurns

File: src/workshed/index.ts [hunk 3]
```diff
@@ -17,14 +18,20 @@ class GarboWorkshed {
...
+        (GarboWorkshed.next ? GarboWorkshed.next.minTurns ?? 0 : 0)
```
Comment (horrible-little-slime) [inline src/workshed/index.ts:33]:
```suggestion
        (GarboWorkshed.next?.minTurns ?? 0)
```

File: src/config.ts [hunk 4]
```diff
@@ -1,30 +1,63 @@
...
+    .map((term) => term[0])
+    .join("");
+  return initials.length >= 3 ? initials : "";
+}
+
+function stripString(s: string): string {
+  if (s.includes(" ")) return stripString(s.replace(" ", ""));
+  else if (s.includes("-")) return stripString(s.replace("-", ""));
+  return s;
+}
```
Comment (horrible-little-slime) [inline src/config.ts:32]: Instead of this bespoke function, use either replaceAll (may need a polyfill?) or a regexp with the `g` flag in `replace`

Comment (horrible-little-slime) [reply src/config.ts:32]: I will probably do something about this Some Day

**Review by horrible-little-slime — APPROVED** [review]

**Pull request merged**